src/daft-plan/src/logical_ops/source.rs

```diff
@@ -84,28 +72,33 @@ impl Source {
                 res.push(format!("Scan op = {}", scan_op));
                 res.push(format!("File schema = {}", source_schema.short_string()));
                 res.push(format!("Partitioning keys = {:?}", partitioning_keys));
-                res.push(format!("Scan pushdowns = {:?}", pushdowns));
+                if let Some(columns) = &pushdowns.columns {
```

Comment:
Should we implement `Display` on `Pushdowns` instead?

Reply:
There are a few reasons I didn't do this as a `Display` implementation:
1. A `Display` implementation returns a single string, rather than a vec of strings. For the tree display mode of the logical plan, I think that we'd want each field to be displayed on its own line rather than be on a single line (many columns in the projection, a long filter expression, etc.) The nice thing about the `multiline_display()` API is that consumers can join them on any separtor they like, be that `"\n"` or `", "`.
2. I was thinking that we'll only want to include the non-null pushdowns, otherwise in the common case we would have a `Pushdowns (columns = None, filters = None, limit = None)` which would clutter the `Source` op section in the repr. We could implement `Display` such that it only includes a field if it's non-null, and returns an empty string if all fields are null, but that seems a bit messy.

We could implement `fn multiline_display()` on `Pushdowns` that returns a `Vec<String>` that both of these match arms could use. How does that sound?

Reply:
That's fine -- should be fine to leave it as-is,

I wish there were simple utilities for displaying hierarchical structs though. Looks like some folks recommend something like `treeline`: https://www.reddit.com/r/rust/comments/od0esb/representing_and_printing_hierarchies/